casper/src/main/scala/coop/rchain/casper/Casper.scala

```diff
@@ -53,11 +51,11 @@ trait Casper[F[_], A] {
   def addBlock(
       b: BlockMessage,
       handleDoppelganger: (BlockMessage, Validator) => F[Unit]
-  ): F[BlockStatus]
+  )(implicit traceId: TraceId): F[BlockStatus]
```

Comment:
Would it be possible to get rid of this parameter here and move it into something like a `Trace` effect (like we do with `Log`) so that this interface (and possibly others) isn't polluted?

Reply:
Agreed, I will continue my analysis on this branch and instead deliver a version that does a ReaderT backed implementation.